tools/ld-chroma-decoder/palcolour.cpp

```diff
@@ -574,29 +583,65 @@ void PalColour::decodeLine(const SourceField &inputField, const ChromaSample *ch
         doYNR(extractedY);
     }
 
-    for (qint32 i = videoParameters.activeVideoStart; i < videoParameters.activeVideoEnd; i++) {
-        double rY = extractedY[i];
-
-        // Scale to 16-bit output
-        rY = qBound(0.0, (rY - videoParameters.black16bIre) * scaledContrast, 65535.0);
-
-        // Rotate the p&q components (at the arbitrary sine/cosine
-        // reference phase) backwards by the burst phase (relative to the
-        // reference phase), in order to recover U and V. The Vswitch is
-        // applied to flip the V-phase on alternate lines for PAL.
-        const double rU =            -(pu[i] * line.bp + qu[i] * line.bq) * scaledSaturation;
-        const double rV = line.Vsw * -(qv[i] * line.bp - pv[i] * line.bq) * scaledSaturation;
-
-        // Convert YUV to RGB, saturating levels at 0-65535 to prevent overflow.
-        // Coefficients from Poynton, "Digital Video and HDTV" first edition, p337 eq 28.6.
-        const double R = qBound(0.0, rY                    + (1.139883 * rV),  65535.0);
-        const double G = qBound(0.0, rY + (-0.394642 * rU) + (-0.580622 * rV), 65535.0);
-        const double B = qBound(0.0, rY + (2.032062 * rU),                     65535.0);
-
-        // Pack the data back into the RGB 16/16/16 buffer
-        const qint32 pp = i * 3; // 3 words per pixel
-        ptr[pp + 0] = static_cast<quint16>(R);
-        ptr[pp + 1] = static_cast<quint16>(G);
-        ptr[pp + 2] = static_cast<quint16>(B);
+    if (configuration.outputYCbCr) {
+        quint16 *ptrY = outputFrame.Y.data() + (((line.number * 2) + inputField.getOffset()) * videoParameters.fieldWidth);
+        quint16 *ptrCb = outputFrame.Cb.data() + (((line.number * 2) + inputField.getOffset()) * videoParameters.fieldWidth);
+        quint16 *ptrCr = outputFrame.Cr.data() + (((line.number * 2) + inputField.getOffset()) * videoParameters.fieldWidth);
+
+        const double yScale = 219.0 * 257.0 / (videoParameters.white16bIre - videoParameters.black16bIre);
+        const double cbScale = 112.0 / 256.0 / (ONE_MINUS_Kb * kB); // Poynton, Eq 25.5 & 28.1
+        const double crScale = 112.0 / 256.0 / (ONE_MINUS_Kr * kR);
+
+        for (qint32 i = videoParameters.activeVideoStart; i < videoParameters.activeVideoEnd; i++) {
+            double rY = extractedY[i];
+
+            // Scale to 16-bit output
+            rY = qBound(1.0*256.0, (rY - videoParameters.black16bIre) * yScale + 16*256, 254.75*256.0);
```

Comment:
Spaces around *. Ditto below.

Reply:
So much whitespace.   All of the *256 mess is to be able to use 8-bit constants instead of magic 16-bit numbers that nobody remembers where they came from.